docs/sequence/PISPTransactionApi.puml

```diff
@@ -71,78 +76,214 @@ rnote right of PISP #Light
   "amountType": "SEND"
 }
 end note
-PISP -> OBS: ""POST /thirdpartyTransactions/{id}/initiate""
+PISP -> 3p_PISP_OBS: POST /thirdpartyTransactions/{id}/initiate
 activate PISP
-activate OBS
-' TODO: line up with state-models.md doc
+activate 3p_PISP_OBS
 
-OBS -> ST8: ""initiate()""
+3p_PISP_OBS -> ST8: initiate()
 activate ST8
-rnote right of ST8: ""state: requestTransaction""
-
-ST8 -> Switch: ""POST /thirdpartyRequests/transactions""
-Switch --> ST8: ""202 Accepted""
-rnote right of ST8: ""state: pendingTransactionAuthorization""
+rnote right of ST8: state: requestTransaction
 
+ST8 -> Switch: POST /thirdpartyRequests/transactions
+activate Switch
+Switch --> ST8: 202 Accepted
+deactivate ST8
+rnote right of ST8: state: pendingTransactionInitiation
 
-note over Switch
-  Switch kicks off the transaction - DFSP issues ""POST /authorizations""
+note over 3p_DFSP_IBS
+  DFSP handles PISP transaction request
 end note
+Switch -> 3p_DFSP_IBS: POST /thirdpartyRequests/transactions
+activate 3p_DFSP_IBS
+3p_DFSP_IBS --> Switch: 202 Accepted
+rnote right of 3p_DFSP_IBS: state: transactionRequestReceived
+note over 3p_DFSP_IBS
+  **transactionRequestState** is **RECEIVED**
+end note
+deactivate Switch
+3p_DFSP_IBS -> Switch: GET /consents/{consentId}
+activate Switch
+Switch --> 3p_DFSP_IBS: 202 Accepted
+deactivate 3p_DFSP_IBS
+Switch -> AUTHS: GET /consents/{consentId}
+activate AUTHS
+AUTHS --> Switch: 202 Accepted
+deactivate Switch
+AUTHS -> AUTHS: lookup for Consent
+AUTHS -> Switch: PUT /consents/{consentId}
+activate Switch
+Switch --> AUTHS: 202 Accepted
+deactivate AUTHS
+Switch -> 3p_DFSP_IBS: PUT /consents/{consentId}
+activate 3p_DFSP_IBS
+3p_DFSP_IBS --> Switch: 202 Accepted
+deactivate Switch
+3p_DFSP_IBS -> 3p_DFSP_IBS: Lookup **consentId, sourceAccountid and pispId**.\nVerify that they exists, and consent is granted with a valid credential
+3p_DFSP_IBS -> 3p_DFSP_IBS: Generate **transactionId**: '1234-1234-1234-1234'
+note over 3p_DFSP_IBS
+  DFSP informs PISP that **transactionId** has been allocated\n
+end note
+rnote right of 3p_DFSP_IBS #Light
+PUT /thirdpartyRequests/transaction/{id}
+{
+  transactionId: '1234-1234-1234-1234',
```

Comment:
Just to clarify: the `transactionId` here is _NOT_ the same as the `{id}` field in the path - that id is referring to the `transactionRequestId` - right?

Reply:
yes, you are right I will fix it properly